libs/ngxtension/reduce-array/src/reduce-array.ts

```diff
@@ -0,0 +1,33 @@
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export function reduceArray<T>(
+	reduceFn: (acc: T, item: T, index: number) => T
+): (source: Observable<T[]>) => Observable<T>;
+
+export function reduceArray<T, R = T>(
+	reduceFn: (acc: R, item: T, index: number) => R,
+	initialValue?: R
```

Comment:
suggestion: `initialValue` here doesn't have to be optional
```suggestion
	initialValue: R
```